controllers/profiles/getOtherProfile.js

```diff
@@ -53,15 +53,25 @@ module.exports = async (req, res) => {
 
     const getFollowerCountResult = getFollowerCount?.[0]?.[0]?.count_follower;
     const getFollowingCountResult = getFollowingCount?.[0]?.[0]?.count_following;
+    const isFollowingResult = isFollowing?.[0]?.length > 0;
 
     const copyUser = {...user.dataValues};
     delete copyUser.following;
     delete copyUser.follower;
-    copyUser.isSignedMessageEnabled = false;
+
+    if (req?.userId) {
+      copyUser.is_following = isFollowingResult;
+    }
+
+    copyUser.isSignedMessageEnabled = true;
     copyUser.isAnonMessageEnabled = false;
-    if (!copyUser.only_received_dm_from_user_following) {
-      copyUser.isSignedMessageEnabled = true;
-      copyUser.isAnonMessageEnabled = copyUser.allow_anon_dm;
+
+    if (copyUser.allow_anon_dm) {
+      if (copyUser.only_received_dm_from_user_following) {
+        copyUser.isAnonMessageEnabled = copyUser.allow_anon_dm && isFollowingResult;
+      } else {
+        copyUser.isAnonMessageEnabled = copyUser.allow_anon_dm;
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for setting `isAnonMessageEnabled` seems correct based on the provided context. However, it might be more readable and maintainable to simplify this nested conditional structure into a single line using logical operators.
```diff
- if (copyUser.allow_anon_dm) {
-   if (copyUser.only_received_dm_from_user_following) {
-     copyUser.isAnonMessageEnabled = copyUser.allow_anon_dm && isFollowingResult;
-   } else {
-     copyUser.isAnonMessageEnabled = copyUser.allow_anon_dm;
-   }
- }
+ copyUser.isAnonMessageEnabled = copyUser.allow_anon_dm && (!copyUser.only_received_dm_from_user_following || isFollowingResult);
```
This change keeps the same logic but makes it easier to understand at a glance.


<!-- This is an auto-generated comment by OSS CodeRabbit -->